### DISCUSSION TITLE
Unifi widget not compatible with Unifi OS Early Access Release ≥ 3.2.5

### DISCUSSION THREAD

**Discussion by dontcrash** [gethomepage/homepage#2280] [Support] [answered]
### Description

![image](https://github.com/gethomepage/homepage/assets/32660358/a0f61782-7f51-467a-b675-13f4a7eedc22)

The default URL Homepage uses results in a "Not Found" page.
I believe at some point Ubiquiti have changed it, this URL for example results in the following JSON
https://172.16.2.1/proxy/network/api/s/default/stat/health

```JSON
{
  "meta": {
    "rc": "ok"
  },
  "data": [
    {
      "subsystem": "wlan",
      "num_user": 4,
      "num_guest": 0,
      "num_iot": 0,
      "tx_bytes-r": 144,
      "rx_bytes-r": 179,
      "status": "warning",
      "num_ap": 1,
      "num_adopted": 2,
      "num_disabled": 0,
      "num_disconnected": 1,
      "num_pending": 0
    },
    {
      "subsystem": "wan",
      "num_gw": 1,
      "num_adopted": 1,
      "num_disconnected": 0,
      "num_pending": 0,
      "status": "ok",
      "wan_ip": "REDACTED",
      "gateways": [],
      "netmask": "255.255.255.0",
      "nameservers": [
        "REDACTED",
        "REDACTED"
      ],
      "num_sta": 20,
      "tx_bytes-r": 2215,
      "rx_bytes-r": 3253,
      "gw_mac": "REDACTED",
      "gw_name": "REDACTED",
      "gw_system-stats": {
        "cpu": "18.0",
        "mem": "68.0",
        "uptime": "792683"
      },
      "gw_version": "3.2.5.13810",
      "isp_name": "REDACTED",
      "isp_organization": "REDACTED",
      "uptime_stats": {
        "WAN": {
          "alerting_monitors": [
            {
              "availability": 100.0,
              "latency_average": 1,
              "target": "1.1.1.1",
              "type": "icmp"
            },
            {
              "availability": 100.0,
              "latency_average": 100,
              "target": "1.1.1.1",
              "type": "dns"
            },
            {
              "availability": 100.0,
              "latency_average": 100,
              "target": "8.8.8.8",
              "type": "dns"
            }
          ],
          "availability": 99.65279999375343,
          "latency_average": 2,
          "monitors": [
            {
              "availability": 100.0,
              "latency_average": 1,
              "target": "www.microsoft.com",
              "type": "icmp"
            },
            {
              "availability": 100.0,
              "latency_average": 21,
              "target": "google.com",
              "type": "icmp"
            },
            {
              "availability": 100.0,
              "latency_average": 1,
              "target": "1.1.1.1",
              "type": "icmp"
            }
          ],
          "time_period": 86400,
          "uptime": 792630
        }
      }
    },
    {
      "subsystem": "www",
      "status": "ok",
      "tx_bytes-r": 2215,
      "rx_bytes-r": 3253,
      "latency": 1,
      "uptime": 160238,
      "drops": 2,
      "xput_up": 33.398,
      "xput_down": 909.11,
      "speedtest_status": "Success",
      "speedtest_lastrun": 1698862833,
      "speedtest_ping": 3,
      "gw_mac": "REDACTED"
    },
    {
      "subsystem": "lan",
      "lan_ip": null,
      "status": "ok",
      "num_user": 16,
      "num_guest": 0,
      "num_iot": 0,
      "tx_bytes-r": 2017,
      "rx_bytes-r": 2062,
      "num_sw": 2,
      "num_adopted": 2,
      "num_disconnected": 0,
      "num_pending": 0
    },
    {
      "subsystem": "vpn",
      "status": "ok",
      "remote_user_enabled": true,
      "remote_user_num_active": 0,
      "remote_user_num_inactive": 0,
      "remote_user_rx_bytes": 1494212,
      "remote_user_tx_bytes": 23008128,
      "remote_user_rx_packets": 9672,
      "remote_user_tx_packets": 23511,
      "site_to_site_enabled": false
    }
  ]
}
```

### Steps to reproduce

Create a local only user on the latest Ubiquiti console
Add the Ubiquiti widget to Homepage

### homepage version

Latest

### Installation method

Docker

### Configuration

```yaml
N/A
```

### Container Logs

N/A

### Browser Logs

N/A

### Troubleshooting

Checked the API URLs from my Ubiquiti console.

### Before submitting, I have made sure to

- [X] Check [the documentation](https://gethomepage.dev/)
- [X] Follow [the troubleshooting guide](https://gethomepage.dev/latest/more/troubleshooting/) (please include output above if applicable).
- [X] Search [existing issues](https://github.com/gethomepage/homepage/search?q=&type=issues) and [discussions](https://github.com/gethomepage/homepage/search?q=&type=discussions).

**Comment by dontcrash** [comment]
I have read the previous issues, read the documentation, completely recreated the homepage container. The issue persists.
I have a UDM-SE running the latest Unifi OS and network application.
This config in services.yaml results in 401.

[2023-11-03T08:13:29.049Z] error: <unifiProxyHandler> HTTP 401 logging in to Unifi. Data: {"error":{"code":401,"message":"Unauthorized"}}

```YAML
    - UDM SE:
        icon: ubiquiti.png
        widget:
          type: unifi
          url: https://172.16.2.1
          username: Homepage
          password: redacted
```

**Comment by shamoon** [comment]
UDMP/UDM SE should return an `x-csrf-token` header, thats how the widget detects which URL to use.
`curl -i -v -k  https://172.16.2.1`

**Comment by dontcrash** [comment]
```console
user@homepage:~$ curl -i -v -k  https://172.16.2.1
*   Trying 172.16.2.1:443...
* Connected to 172.16.2.1 (172.16.2.1) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS header, Finished (20):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.2 (OUT), TLS header, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=unifi.local
*  start date: Mar 26 08:13:34 2022 GMT
*  expire date: Jun 28 08:13:34 2024 GMT
*  issuer: CN=unifi.local
*  SSL certificate verify result: self-signed certificate (18), continuing anyway.
* Using HTTP2, server supports multiplexing
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* Using Stream ID: 1 (easy handle 0x560da85cbe90)
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
> GET / HTTP/2
> Host: 172.16.2.1
> user-agent: curl/7.81.0
> accept: */*
> 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
< HTTP/2 200 
HTTP/2 200 
< server: nginx
server: nginx
< date: Fri, 03 Nov 2023 09:11:45 GMT
date: Fri, 03 Nov 2023 09:11:45 GMT
< content-type: text/html
content-type: text/html
< content-length: 826
content-length: 826
< last-modified: Tue, 24 Oct 2023 12:20:43 GMT
last-modified: Tue, 24 Oct 2023 12:20:43 GMT
< etag: "6537b69b-33a"
etag: "6537b69b-33a"
< expires: Fri, 03 Nov 2023 09:11:44 GMT
expires: Fri, 03 Nov 2023 09:11:44 GMT
< cache-control: no-cache
cache-control: no-cache
< access-control-allow-credentials: false
access-control-allow-credentials: false
< access-control-expose-headers: Content-Disposition, Content-Range, Filename, Location, Range, Upload-Length, Upload-Offset, X-Connection-Type, X-Csrf-Token, X-File-Id, X-Token-Expire-Time, X-Updated-Csrf-Token
access-control-expose-headers: Content-Disposition, Content-Range, Filename, Location, Range, Upload-Length, Upload-Offset, X-Connection-Type, X-Csrf-Token, X-File-Id, X-Token-Expire-Time, X-Updated-Csrf-Token
< referrer-policy: no-referrer
referrer-policy: no-referrer
< strict-transport-security: max-age=15552000; includeSubDomains
strict-transport-security: max-age=15552000; includeSubDomains
< x-content-type-options: nosniff
x-content-type-options: nosniff
< x-dns-prefetch-control: off
x-dns-prefetch-control: off
< x-frame-options: SAMEORIGIN
x-frame-options: SAMEORIGIN
< x-xss-protection: 1; mode=block
x-xss-protection: 1; mode=block
< accept-ranges: bytes
accept-ranges: bytes

< 
* Connection #0 to host 172.16.2.1 left intact
<!doctype html><html lang="en"><head><meta charset="utf-8"><link rel="icon" href="/favicon.ico?v3" sizes="any"><link rel="icon" href="/favicon.svg?v3" type="image/svg+xml"><link rel="apple-touch-icon" href="/apple-touch-icon.png?v3"><title>UniFi OS</title><meta name="viewport" content="width=device-width,initial-scale=1"><script>window.UNIFI_OS_MANIFEST = {"model":{"id":"UDMPROSE","shortName":"UDM SE","longName":"UniFi Dream Machine SE"},"images":{"small":"/assets/images/small.png?udmprose-2.0.0","medium":"/assets/images/medium.png?udmprose-2.0.0","large":"/assets/images/large.png?udmprose-2.0.0","huge":"/assets/images/huge.png?udmprose-2.0.0"}}</script><script defer="defer" src="/main.e8aacf3d73c3fd2b5bbf.js"></script><link href="/main.0e9197a3.css" rel="stylesheet"></head><body><div id="root"></div></body></html>
```

**Comment by shamoon** [comment]
So thats a UDMSE and what version of the OS / network is that running?

**Comment by dontcrash** [comment]
This is happening on the latest release, OS 3.2.5, network 7.5.187

https://community.ui.com/releases/UniFi-OS-Dream-Machine-SE-3-2-5/c2cb12bc-1090-4546-94e5-823d005e33bc
https://community.ui.com/releases/UniFi-Network-Application-7-5-187/408b64c5-a485-4a37-843c-31e87140be64

**Comment by dontcrash** [comment]
**Edit: After looking at the source, I can see there are checks for if it's a pro device: const udmpPrefix = "/proxy/network";
Thinking Ubiquti changed something**


I believe I have found the root cause, any "pro" controller needs to have "/proxy/network/" before "api" in the URI

https://ubntwiki.com/products/software/unifi-controller/api#udm_pro_api

![image](https://github.com/gethomepage/homepage/assets/32660358/e18733b2-f24b-46fe-9099-3433dc5ca10a)

**Comment by dontcrash** [comment]
If anyone else stumbles upon this, I ended up creating a customapi widget using a token I got from web inspector.
Am unsure how long the token is valid for, will find out in the coming days/weeks!

![image](https://github.com/gethomepage/homepage/assets/32660358/c2b99008-d555-4f98-ab9d-9edb402e1fc8)

services.yaml
```YAML
- Statistics:
    - UDM SE:
        icon: ubiquiti.png
        widget:
          type: customapi
          url: https://172.16.2.1/proxy/network/api/s/default/stat/health
          headers:
            Cookie: 'TOKEN=abc123'  
          mappings:
            - field:
                data:
                  1: wan_ip
              label: WAN
              format: text
            - field:
                data:
                  1:
                    gw_system-stats: uptime
              label: Uptime
              format: number
              suffix: Hours
              scale: 0.000277778
            - field:
                data:
                  1:
                    gw_system-stats: cpu
              label: CPU
              format: number
              suffix: '%'
            - field:
                data:
                  1:
                    gw_system-stats: mem
              label: RAM
              format: number
              suffix: '%'
```

**Reply by therus000** [reply]
Thanx
but how to  get this value
          headers:
            Cookie: 'TOKEN=abc123'  ?

**Comment by shamoon** [comment]
> I believe I have found the root cause, any "pro" controller needs to have "/proxy/network/" before "api" in the URI

Yes, this is exactly the point I was making [above](https://github.com/gethomepage/homepage/issues/2269#issuecomment-1792040512), homepage "detects" whether the controller is Pro (or SE, they have always been the same in this regard) by checking for the presence of the `x-csrf-token` token
https://github.com/gethomepage/homepage/blob/5512d05f00026504d73f0230436b40788028eb4e/src/widgets/unifi/proxy.js#L83-L86

So I'm running Network 7.5.187 but OS v3.1.16, works fine. 3.2.5 seems like early access release? The thing is we dont know:

1. If its Unifi OS > 3.1.16 that does this (I would've guessed it was the Network app, but maybe not) or its UDMSE vs UDMP
2. If this will change before the next unifi OS is actually released

So tldr, I'm going to keep an eye out for this, an alternative would be to add a way to manually override this which might be the longterm solution if the unifi API is going to have breaking changes like this, or there may be an alternative way to sniff udmp vs controller

Will update this thread if / when have more to say.

**Comment by Magelo** [comment]
I have the exact same issue.

![image](https://github.com/gethomepage/homepage/assets/10362303/44fe3bad-f66c-4329-b660-297d1799f474)

I have Network 8.0.2 and UnifiOS 3.2.5. I think UnifiOS 3.2.5 broke Unifi widget.

If I use this URL : https://192.168.1.1/proxy/network/api/stat/sites instead of https://192.168.1.1/api/stat/sites I receive a JSON answer.

**Comment by shamoon** [comment]
Would you also mind posting the output of curl -i -v -k https://192.168.1.1

**Comment by Magelo** [comment]
```
curl -i -v -k https://192.168.1.1
*   Trying 192.168.1.1:443...
* Connected to 192.168.1.1 (192.168.1.1) port 443 (#0)
* ALPN: offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=unifi.local
*  start date: Oct 23 16:36:33 2023 GMT
*  expire date: Jan 25 17:36:33 2026 GMT
*  issuer: CN=unifi.local
*  SSL certificate verify result: self-signed certificate (18), continuing anyway.
* using HTTP/2
* h2h3 [:method: GET]
* h2h3 [:path: /]
* h2h3 [:scheme: https]
* h2h3 [:authority: 192.168.1.1]
* h2h3 [user-agent: curl/7.88.1]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x5596cac760)
> GET / HTTP/2
> Host: 192.168.1.1
> user-agent: curl/7.88.1
> accept: */*
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
< HTTP/2 200
HTTP/2 200
< server: nginx
server: nginx
< date: Tue, 07 Nov 2023 18:08:09 GMT
date: Tue, 07 Nov 2023 18:08:09 GMT
< content-type: text/html
content-type: text/html
< content-length: 818
content-length: 818
< last-modified: Mon, 23 Oct 2023 16:36:33 GMT
last-modified: Mon, 23 Oct 2023 16:36:33 GMT
< etag: "6536a111-332"
etag: "6536a111-332"
< expires: Tue, 07 Nov 2023 18:08:08 GMT
expires: Tue, 07 Nov 2023 18:08:08 GMT
< cache-control: no-cache
cache-control: no-cache
< access-control-allow-credentials: false
access-control-allow-credentials: false
< access-control-expose-headers: Content-Disposition, Content-Range, Filename, Location, Range, Upload-Length, Upload-Offset, X-Connection-Type, X-Csrf-Token, X-File-Id, X-Token-Expire-Time, X-Updated-Csrf-Token
access-control-expose-headers: Content-Disposition, Content-Range, Filename, Location, Range, Upload-Length, Upload-Offset, X-Connection-Type, X-Csrf-Token, X-File-Id, X-Token-Expire-Time, X-Updated-Csrf-Token
< referrer-policy: no-referrer
referrer-policy: no-referrer
< strict-transport-security: max-age=15552000; includeSubDomains
strict-transport-security: max-age=15552000; includeSubDomains
< x-content-type-options: nosniff
x-content-type-options: nosniff
< x-dns-prefetch-control: off
x-dns-prefetch-control: off
< x-frame-options: SAMEORIGIN
x-frame-options: SAMEORIGIN
< x-xss-protection: 1; mode=block
x-xss-protection: 1; mode=block
< accept-ranges: bytes
accept-ranges: bytes

<
* Connection #0 to host 192.168.1.1 left intact
<!doctype html><html lang="en"><head><meta charset="utf-8"><link rel="icon" href="/favicon.ico?v3" sizes="any"><link rel="icon" href="/favicon.svg?v3" type="image/svg+xml"><link rel="apple-touch-icon" href="/apple-touch-icon.png?v3"><title>UniFi OS</title><meta name="viewport" content="width=device-width,initial-scale=1"><script>window.UNIFI_OS_MANIFEST = {"model":{"id":"UDMPRO","shortName":"UDM Pro","longName":"UniFi Dream Machine Pro"},"images":{"small":"/assets/images/small.png?udmpro-2.0.0","medium":"/assets/images/medium.png?udmpro-2.0.0","large":"/assets/images/large.png?udmpro-2.0.0","huge":"/assets/images/huge.png?udmpro-2.0.0"}}</script><script defer="defer" src="/main.e8aacf3d73c3fd2b5bbf.js"></script><link href="/main.0e9197a3.css" rel="stylesheet"></head><body><div id="root"></div></body></html>p
```

**Comment by shamoon** [comment]
Thanks.

So the question remains whether the final UnifiOS will have this change but in general I think installing Unifi pre-release software is very risky so I'm not going to do it. That 3.2.5 release was in fact pulled because it borked people's internet: https://community.ui.com/releases/UniFi-OS-Dream-Machine-SE-3-2-5/c2cb12bc-1090-4546-94e5-823d005e33bc#comment/10fa67a1-668e-473e-b1c7-766d73a18252

Then there is the fun complication that the Unifi controller image is going to be deprecated: https://info.linuxserver.io/issues/2023-09-06-unifi-controller/

Basically will look at alternative means of auto-detection and / or manual specifying in the config, maybe both

**Reply by dontcrash** [reply]
No one knows what Ubiquiti are going to do next 😅
I would vote for the option to manually override the URL as an optional parameter.

**Comment by shamoon** [comment, marked as answer]
If either of you would like to pull the `main` image and give that a try, let me know. Thanks

**Reply by Magelo** [reply]
`main` is working for me with Network 8.0.6 and UnifiOS 3.2.5 on a UniFi Dream Machine Pro. Thanks.

**Reply by shamoon** [reply]
Awesome. Hopefully things a durable fix, I suppose we’ll see…

**Reply by thefirstnoe1** [reply]
Working for me as well. Thank you for the fix!

**Comment by ray73864** [comment]
Having an issue with my CloudKey Gen 2.

I am running UnifiOS 3.2.7, and no matter how much I try, all I can get out of it is 401 Unauthorized.

Network version is 8.0.24.

Currently using Homepage version 0.8.3 as that is what TrueCharts has.

```
API Error: HTTP Error 401
URL: https://192.168.234.20/api/stat/sites
Response Data:
{"error":{"code":401,"message":"Unauthorized"}}
```

If I perform that 'curl' command, I get:

```
*   Trying 192.168.234.20:443...
* Connected to 192.168.234.20 (192.168.234.20) port 443
* schannel: disabled automatic use of client certificate
* schannel: using IP address, SNI is not supported by OS.
* ALPN: curl offers http/1.1
* ALPN: server accepted http/1.1
* using HTTP/1.1
> GET / HTTP/1.1
> Host: 192.168.234.20
> User-Agent: curl/8.4.0
> Accept: */*
>
* schannel: remote party requests renegotiation
* schannel: renegotiating SSL/TLS connection
* schannel: SSL/TLS connection renegotiated
* schannel: remote party requests renegotiation
* schannel: renegotiating SSL/TLS connection
* schannel: SSL/TLS connection renegotiated
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
< Server: nginx
Server: nginx
< Date: Fri, 29 Dec 2023 02:37:36 GMT
Date: Fri, 29 Dec 2023 02:37:36 GMT
< Content-Type: text/html
Content-Type: text/html
< Content-Length: 807
Content-Length: 807
< Last-Modified: Thu, 14 Dec 2023 16:21:25 GMT
Last-Modified: Thu, 14 Dec 2023 16:21:25 GMT
< Connection: keep-alive
Connection: keep-alive
< ETag: "657b2b85-327"
ETag: "657b2b85-327"
< Expires: Fri, 29 Dec 2023 02:37:35 GMT
Expires: Fri, 29 Dec 2023 02:37:35 GMT
< Cache-Control: no-cache
Cache-Control: no-cache
< Access-Control-Allow-Credentials: false
Access-Control-Allow-Credentials: false
< Access-Control-Expose-Headers: Content-Disposition, Content-Range, Filename, Location, Range, Upload-Length, Upload-Offset, X-Connection-Type, X-Csrf-Token, X-File-Id, X-Token-Expire-Time, X-Updated-Csrf-Token
Access-Control-Expose-Headers: Content-Disposition, Content-Range, Filename, Location, Range, Upload-Length, Upload-Offset, X-Connection-Type, X-Csrf-Token, X-File-Id, X-Token-Expire-Time, X-Updated-Csrf-Token
< Referrer-Policy: no-referrer
Referrer-Policy: no-referrer
< Strict-Transport-Security: max-age=15552000; includeSubDomains
Strict-Transport-Security: max-age=15552000; includeSubDomains
< X-Content-Type-Options: nosniff
X-Content-Type-Options: nosniff
< X-DNS-Prefetch-Control: off
X-DNS-Prefetch-Control: off
< X-Frame-Options: SAMEORIGIN
X-Frame-Options: SAMEORIGIN
< X-XSS-Protection: 1; mode=block
X-XSS-Protection: 1; mode=block
< Accept-Ranges: bytes
Accept-Ranges: bytes

<
<!doctype html><html lang="en"><head><meta charset="utf-8"><link rel="icon" href="/favicon.ico?v3" sizes="any"><link rel="icon" href="/favicon.svg?v3" type="image/svg+xml"><link rel="apple-touch-icon" href="/apple-touch-icon.png?v3"><title>UniFi OS</title><meta name="viewport" content="width=device-width,initial-scale=1"><script>window.UNIFI_OS_MANIFEST = {"model":{"id":"UCKG2","shortName":"UCK G2","longName":"UniFi Cloud Key G2"},"images":{"small":"/assets/images/small.png?uckg2-2.0.0","medium":"/assets/images/medium.png?uckg2-2.0.0","large":"/assets/images/large.png?uckg2-2.0.0","huge":"/assets/images/huge.png?uckg2-2.0.0"}}</script><script defer="defer" src="/main.8aa98e7b9c40b76fd5bf.js"></script><link href="/main.0e9197a3.css" rel="stylesheet"></head><body><div id="root"></div></body></html>* Connection #0 to host 192.168.234.20 left intact
```

**Reply by shamoon** [reply]
re-create (not just restart) the container

**Reply by ray73864** [reply]
I did.

Didn't help.

**Reply by ray73864** [reply]
I also found that the correct URL is /proxy/network before the /api/stat/sites, but that still resulted in a 401 Unauthorised.

If I manually nagivate to https://192.168.234.20/proxy/network/api/stat/sites, I get a listing, but as soon as it tries to use a username/password, it fails.

**Reply by ray73864** [reply]
I can't even curl it locally from the command line outside of the TrueNAS Scale pod, just on my local computer, comes back with the same unauthorised issue :(

**Comment by fliebe92** [comment]
I'm facing the following issue:

```
API Error: HTTP Error 401
URL: https://10.19.92.1/api/stat/sites
Response Data:
{"error":{"code":401,"message":"Unauthorized"}}
```

Recreating the homepage container did not help. Any idea?

UniFi Dream Router
* UniFi OS 3.2.9 
* Network 8.0.28

```
curl -i -v -k https://10.19.92.1

*   Trying 10.19.92.1:443...
* Connected to 10.19.92.1 (10.19.92.1) port 443
* ALPN: curl offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384 / X25519 / RSASSA-PSS
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=<redacted>
*  start date: Jan  8 09:00:24 2024 GMT
*  expire date: Apr  7 09:00:23 2024 GMT
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
*   Certificate level 0: Public key type RSA (2048/112 Bits/secBits), signed using sha256WithRSAEncryption
*   Certificate level 1: Public key type RSA (2048/112 Bits/secBits), signed using sha256WithRSAEncryption
*   Certificate level 2: Public key type RSA (4096/152 Bits/secBits), signed using sha256WithRSAEncryption
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://10.19.92.1/
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: 10.19.92.1]
* [HTTP/2] [1] [:path: /]
* [HTTP/2] [1] [user-agent: curl/8.5.0]
* [HTTP/2] [1] [accept: */*]
> GET / HTTP/2
> Host: 10.19.92.1
> User-Agent: curl/8.5.0
> Accept: */*
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
< HTTP/2 200
HTTP/2 200
< server: nginx
server: nginx
< date: Sat, 03 Feb 2024 21:18:32 GMT
date: Sat, 03 Feb 2024 21:18:32 GMT
< content-type: text/html
content-type: text/html
< content-length: 794
content-length: 794
< last-modified: Tue, 16 Jan 2024 02:21:11 GMT
last-modified: Tue, 16 Jan 2024 02:21:11 GMT
< etag: "65a5e817-31a"
etag: "65a5e817-31a"
< expires: Sat, 03 Feb 2024 21:18:31 GMT
expires: Sat, 03 Feb 2024 21:18:31 GMT
< cache-control: no-cache
cache-control: no-cache
< access-control-allow-credentials: false
access-control-allow-credentials: false
< access-control-expose-headers: Content-Disposition, Content-Range, Filename, Location, Range, Upload-Length, Upload-Offset, X-Connection-Type, X-Csrf-Token, X-File-Id, X-Token-Expire-Time, X-Updated-Csrf-Token
access-control-expose-headers: Content-Disposition, Content-Range, Filename, Location, Range, Upload-Length, Upload-Offset, X-Connection-Type, X-Csrf-Token, X-File-Id, X-Token-Expire-Time, X-Updated-Csrf-Token
< referrer-policy: no-referrer
referrer-policy: no-referrer
< strict-transport-security: max-age=15552000; includeSubDomains
strict-transport-security: max-age=15552000; includeSubDomains
< x-content-type-options: nosniff
x-content-type-options: nosniff
< x-dns-prefetch-control: off
x-dns-prefetch-control: off
< x-frame-options: SAMEORIGIN
x-frame-options: SAMEORIGIN
< x-xss-protection: 1; mode=block
x-xss-protection: 1; mode=block
< accept-ranges: bytes
accept-ranges: bytes

<
* Connection #0 to host 10.19.92.1 left intact
<!doctype html><html lang="en"><head><meta charset="utf-8"><link rel="icon" href="/favicon.ico?v3" sizes="any"><link rel="icon" href="/favicon.svg?v3" type="image/svg+xml"><link rel="apple-touch-icon" href="/apple-touch-icon.png?v3"><title>UniFi OS</title><meta name="viewport" content="width=device-width,initial-scale=1"><script>window.UNIFI_OS_MANIFEST = {"model":{"id":"UDR","shortName":"UDR","longName":"UniFi Dream Router"},"images":{"small":"/assets/images/small.png?udr-2.0.0","medium":"/assets/images/medium.png?udr-2.0.0","large":"/assets/images/large.png?udr-2.0.0","huge":"/assets/images/huge.png?udr-2.0.0"}}</script><script defer="defer" src="/main.a61084f6cbcc2d31593e.js"></script><link href="/main.0e9197a3.css" rel="stylesheet"></head><body><div id="root"></div></body></html>
```

**Reply by shamoon** [reply]
Please open a new thread
